railties/test/application/rake/migrations_test.rb

```diff
@@ -431,6 +431,180 @@ class TwoMigration < ActiveRecord::Migration::Current
+          assert_not File.exist?("db/snowflake_schema.rb"), "should not dump schema when configured not to"
+        end
+      end
+
+      test "single db schema generation when dump_schema_after_migration is false and schema_dump is true" do
+        add_to_config("config.active_record.dump_schema_after_migration = false")
+
+        app_file "config/database.yml", <<~EOS
+          development:
+            adapter: sqlite3
+            database: 'dev_db'
+            schema_dump: true
+        EOS
+
+        Dir.chdir(app_path) do
+          rails "generate", "model", "book", "title:string"
+          rails "db:migrate"
+
+          assert_not File.exist?("db/schema.rb"), "should not dump schema when configured not to"
+        end
+      end
+
+      test "multi db schema generation when dump_schema_after_migration is false and schema_dump is true" do
+        add_to_config("config.active_record.dump_schema_after_migration = false")
+
+        app_file "config/database.yml", <<~EOS
+          development:
+            primary:
+              adapter: sqlite3
+              database: 'dev_db'
+              schema_dump: true
+            snowflake:
+              adapter: sqlite3
+              database: 'dev_db'
+              schema_dump: true
+        EOS
+
+        Dir.chdir(app_path) do
+          rails "generate", "model", "book", "title:string"
+          rails "db:migrate"
+
+          assert_not File.exist?("db/schema.rb"), "should not dump schema when configured not to"
+          assert_not File.exist?("db/snowflake_schema.rb"), "should not dump schema when configured not to"
+        end
+      end
+
+      test "multi db schema generation with schema dump only for primary" do
+        add_to_config("config.active_record.dump_schema_after_migration = true")
+
+        app_file "config/database.yml", <<~EOS
+          development:
+            primary:
+              adapter: sqlite3
+              database: 'primary_dev_db'
+              schema_dump: true
+            snowflake:
+              adapter: sqlite3
+              database: 'snowflake_dev_db'
+              schema_dump: false
+        EOS
+
+        Dir.chdir(app_path) do
+          rails "generate", "model", "book", "title:string"
+          rails "db:migrate:primary", "db:migrate:snowflake"
+
+          assert File.exist?("db/schema.rb"), "should not dump schema when configured not to"
+          assert_not File.exist?("db/snowflake_schema.rb"), "should not dump schema when configured not to"
+        end
+      end
+
+      test "multi db schema generation with schema dump only for snowflake" do
+        add_to_config("config.active_record.dump_schema_after_migration = true")
+
+          app_file "config/database.yml", <<~EOS
+            development:
+              primary:
+                adapter: sqlite3
+                database: primary_dev_db
+                schema_dump: false
+              snowflake:
```

Comment:
Can you name this to something else other than `snowflake`? `snowflake` is often used as an insult in the US.